x/protocolpool/keeper/keeper.go

```diff
@@ -397,19 +393,23 @@ func (k Keeper) calculateClaimableFunds(ctx context.Context, recipient sdk.AccAd
 	periodsPassed := int64(timeElapsed) / int64(*budget.Period)
 
 	// Calculate the amount to distribute for all passed periods
-	coinsToDistribute := math.NewInt(periodsPassed).Mul(budget.TotalBudget.Amount.QuoRaw(int64(budget.Tranches)))
+	coinsToDistribute := math.NewInt(periodsPassed).Mul(budget.BudgetPerTranche.Amount)
 	amount = sdk.NewCoin(budget.TotalBudget.Denom, coinsToDistribute)
 
 	// update the budget's remaining tranches
-	budget.TranchesLeft -= uint64(periodsPassed)
+	if budget.TranchesLeft > uint64(periodsPassed) {
+		budget.TranchesLeft -= uint64(periodsPassed)
+	} else {
+		budget.TranchesLeft = 0
+	}
```

Comment:
I think so. The calculation of `periodsPassed` is based on time elapsed since the last claim time. If there's a halt in the chain, the time elapsed won't progress, and therefore, those periods won't be counted.

Reply:
Looking at it again, I don't think this is the case. We do `coinsToDistribute := math.NewInt(periodsPassed).Mul(budget.BudgetPerTranche.Amount)`, so if the chain halts and restarts with the correct time, we'd be paying the correct amounts.

Now another question is, what happens if this gets called way after? Let's say there are 2 tranches left but this gets called after 3 periods. What's stopping it from sending more? Maybe it's somewhere else in the code but I couldn't find it

Reply:
Fixed with this commit: https://github.com/cosmos/cosmos-sdk/pull/20154/commits/5a0ddd6353f8f440eb6f417ffaa494f18c792e7f

Now periods passed are capped to the tranches left